tests/tasks/app/test_job_control.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+import io
+from unittest import mock
+import tests.extensions.tus.utils as tus_utils
+import tests.modules.asset_groups.resources.utils as asset_group_utils
+
+from invoke import MockContext
+
+
+# Check that the task methods for the asset control job tasks print the correct output
+def test_asset_group_detection_jobs(
+    flask_app, flask_app_client, researcher_1, staff_user, test_root, db
+):
+    # pylint: disable=invalid-name
+    from tests.modules.asset_groups.resources.utils import TestCreationData
+    from tests import utils as test_utils
+
+    orig_objs = test_utils.all_count(db)
+
+    transaction_id, test_filename = tus_utils.prep_tus_dir(test_root)
+    asset_group_uuid = None
+    try:
+        data = TestCreationData(transaction_id)
+        data.add_filename(0, test_filename)
+        data.set_field('speciesDetectionModel', ['someSortOfModel'])
+
+        # Simulate a valid response from Sage but don't actually send the request to Sage
+        with mock.patch.object(
+            flask_app.acm,
+            'request_passthrough_result',
+            return_value={'success': True},
+        ):
+            resp = asset_group_utils.create_asset_group(
+                flask_app_client, None, data.get()
+            )
+            asset_group_uuid = resp.json['guid']
+
+        # Now see that the task gets what we expect
+        with mock.patch('app.create_app'):
```

Comment:
@karenc this was copied from your example but while I understand why everything else is required, I don't see what this line is doing. I do see that without it "Bad" things happen

Reply:
What this is doing is patching `app.create_app` so that it doesn't try to create another app because we already have a `flask_app` fixture.  Weird stuff happens when another app is created so I patch `app.create_app`.